test/staged.jl

```diff
@@ -250,3 +250,25 @@ end
 @test f22440(0.0) === f22440kernel(0.0)
 @test f22440(0.0f0) === f22440kernel(0.0f0)
 @test f22440(0) === f22440kernel(0)
+
+# PR #23168
+
+function f23168(a, x)
+    push!(a, 1)
+    if @generated
+        :(y = (x + x, $x))
+    else
+        y = (2x, typeof(x))
+    end
+    push!(a, 2)
+    return y
```

Comment:
Can we have a test/do we support functions with multiple generated blocks?

Reply:
Yes, multiple generated blocks should work. You just get two versions, one with the `then` part of each generated block selected, and one with the `else` parts.